util/pagination/fields.js

```diff
@@ -0,0 +1,574 @@
+    const oldPage = this.page;
+    this.setPage(param);
+
+    if (oldPage === 1 || oldPage === this.pages || this.page === 1 || this.page === this.pages) {
+      await this.clientMessage.message.reactions.removeAll();
+
+      this._loadList(true);
+    } else {
+      await this._loadList(false);
+
+      this._awaitResponse();
+    }
+  }
+
+  /**
+   * Awaits the reaction from the user.
+   * @private
+   * @protected
+   */
+  async _awaitResponse() {
+    const emojis = Object.values(this.emojis);
+    const filter = (r, u) => {
+      if (this.authorisedUser)
+        return u.id === this.authorisedUser.id && emojis.includes(r.emoji.name);
+
+      return !u.bot && emojis.includes(r.emoji.name);
+    };
+    const clientMessage = this.clientMessage.message;
+
+    try {
+      const responses = await clientMessage.awaitReactions(filter, { max: 1, time: this.timeout, errors: ["time"] });
+      const response = responses.first();
+      const user = response.users.last();
+      const emoji = response.emoji.name;
+
+      if (emoji === this.emojis.delete) return clientMessage.delete();
+
+      await response.users.remove(user);
+
+      switch (emoji) {
+        case this.emojis.back:
+          if (this.page === 1) return this._awaitResponse();
+
+          this._loadPage("back");
+          break;
+
+        case this.emojis.jump:
+          if (this.pages <= 2) return this._awaitResponse();
+
+          this._awaitResponseEx(user);
+          break;
+
+        case this.emojis.forward:
+          if (this.page === this.pages) return this._awaitResponse();
+
+          this._loadPage("forward");
+          break;
+      }
+    } catch (c) {
+      clientMessage.reactions.removeAll().catch(err => {
+        throw err;
+      });
+
+      if (c instanceof Error) throw c;
+    }
+  }
+
+  /**
+   * Awaits the custom page input from the user.
+   * @private
+   * @protected
+   * @param {User} user - The user who reacted to jump on a certain page.
+   */
+  async _awaitResponseEx(user) {
+    const filter = m => {
+      const supposedPage = parseInt(m.content);
+
+      return (
+        m.author.id === user.id && (
+          (!isNaN(m.content) && supposedPage !== this.page && supposedPage >= 1 && supposedPage <= this.pages) ||
+          m.content.toLowerCase() === "cancel"
+        )
+      );
+    };
+    const channel = this.clientMessage.message.channel;
+    const prompt = await channel.send("To what page would you like to jump? Say `cancel` to cancel the prompt.");
+
+    try {
+      const responses = await channel.awaitMessages(filter, { max: 1, time: this.timeout, errors: ["time"] });
+      const response = responses.first();
+      let content = response.content;
+
+      await prompt.delete();
+      await response.delete();
+
+      if (content === "cancel") return this._awaitResponse();
+
+      content = parseInt(content);
+
+      this._loadPage(content);
+    } catch (c) {
+      prompt.delete().catch(err => {
```

Comment:
Missed await, in this case this promise throws, this will throw an unhanded error. Use the following code instead of re-throwing. Same with all the occurrences that follow this code pattern.

```javascript
await prompt.delete();
```